### PR TITLE
add build for Python 3.11 for linux

### LINKED ISSUE #1855: build a standalone executable for linux using python 3.11/12
python 3.11 is about 15% faster than python 3.8 for capa workloads:

Screenshot 2023-11-14 1 44 58 PM

Screenshot 2023-11-14 2 04 58 PM

so, provide a standalone executable that can be widely deployed that includes these performance optimizations. there's still value in distributing the standalone executables built on python 3.8 since they work on more platforms. but for environments that support python 3.11, the performance improvements are basically "free".

so, update the GH action workflow to build this executable and upload to the release pages, using a filename indicating its based on python-311.

### PR DESCRIPTION
**Pull request by mr-tz**
closes #1855

### Checklist

- [x] No CHANGELOG update needed

- [ ] No new tests needed

- [ ] No documentation update needed

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -55,13 +62,17 @@ jobs:
         run: dist/capa "tests/data/499c2a85f6e8142c3f48d4251c9c7cd6.raw32"
       - name: Does it run (ELF)?
         run: dist/capa "tests/data/7351f8a40c5450557b24622417fc478d.elf_"
+      - name: Does it run (CAPE)?
+        run: |
+          7z e "tests/data/dynamic/cape/v2.2/d46900384c78863420fb3e297d0a2f743cd2b6b3f7f82bf64059a168e07aceb7.json.gz"
```
Comment (williballenthin) [inline .github/workflows/build.yml:67]: ah that's cool

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -11,34 +11,41 @@ permissions:
...
+        # using Python 3.8 to support running across multiple operating systems including Windows 7
         include:
           - os: ubuntu-20.04
             # use old linux so that the shared library versioning is more portable
             artifact_name: capa
             asset_name: linux
+            python_version: 3.8
+          - os: ubuntu-20.04
+            artifact_name: capa
+            asset_name: linux-py311
```
Comment (williballenthin) [inline .github/workflows/build.yml:29]: i really like the way you did this, to keep the default linux one named the same and the optimized one available if desired.

**Review by williballenthin — APPROVED** [review]

**Comment by williballenthin** [comment]
we can manually build a v6.1.0 version in the meantime until v7 is released.

**Comment by mr-tz** [comment]
via the `edited` event we may even be able to add it automatically to the existing release (by updating the text, e.g.)

**Comment by mr-tz** [comment]
https://github.com/mandiant/capa/actions/runs/7038874727

**Comment by mr-tz** [comment]
using https://github.com/mandiant/capa/actions/runs/7039024510 to build it

**Pull request merged**